salt/client/ssh/__init__.py

```diff
@@ -1401,8 +1421,15 @@ def categorize_shim_errors(self, stdout_bytes, stderr_bytes, retcode):
             ),
             (
                 (salt.defaults.exitcodes.EX_THIN_PYTHON_INVALID,),
-                'Python interpreter is too old',
-                'salt requires python 2.6 or newer on target hosts, must have same major version as origin host'
+                'Python version mismatched',
+                python_mismatch_options.get(
+                    # check for major.minor first
+                    '%s.%s' % sys.version_info[0:2]
+                ) or python_mismatch_options.get(
+                    # check for major.any
+                    '%s.x' % sys.version_info[0]
+                    # show the default
+                ) or python_mismatch_options['default']
```

Comment:
Please do not use `'%s.%s' % foo`, but `'{}.{}'.format(...)` instead, as per internal Salt agreements. As well as we probably do not need "3.x" but can be just "3". Also I would do it less complex, but use `.get(...)` as intended:
```python
py_mismatch.get('{}.{}'.format(*sys.version_info[:2]),
                py_mismatch.get(str(sys.version_info[0]),
                py_mismatch['default']))```